Library/QuickPulseStateManager.ts

```diff
@@ -124,14 +125,15 @@ class QuickPulseStateManager {
         }
 
         let currentTimeout = QuickPulseStateManager._isCollectingData ? 1000 : 5000;
-        if (QuickPulseStateManager._isCollectingData && Date.now() - this._lastSuccessTime >= 20000) {
+        if (QuickPulseStateManager._isCollectingData && Date.now() - this._lastSuccessTime >= 20000 && !this._lastSendSucceeded) {
             // Haven't posted successfully in 20 seconds, so wait 60 seconds and ping
             QuickPulseStateManager._isCollectingData = false;
             currentTimeout = 60000;
-        } else if (!QuickPulseStateManager._isCollectingData && Date.now() - this._lastSuccessTime >= 60000) {
+        } else if (!QuickPulseStateManager._isCollectingData && Date.now() - this._lastSuccessTime >= 60000 && !this._lastSendSucceeded) {
```

Comment:
Should the 60s be in a variable somewhere since it's repeated in a few places?

Reply:
Added static consts